.github/actions/setup-common/action.yml

```diff
@@ -0,0 +1,52 @@
+name: 'Common setup for all OSes'
+description: 'Installs common dependencies'
+inputs:
+  codec-aom:
+    description: 'Can take the values: OFF, LOCAL, SYSTEM'
+    default: 'OFF'
+  codec-dav1d:
+    description: 'Can take the values: OFF, LOCAL, SYSTEM'
+    default: 'OFF'
+  codec-rav1e:
+    description: 'Can take the values: OFF, LOCAL, SYSTEM'
+    default: 'OFF'
+  recent-cmake:
+    description: 'Can take the values: true, false. Only useful on Linux'
+    default: 'false'
+runs:
+  using: "composite"
+  steps:
+    - name: Set up Python
+      uses: actions/setup-python@82c7e631bb3cdc910f68e0081d67478d79c6982d # v5.1.0
+      with:
+        python-version: '3.x'
+    - name: Set up CMake < 3.18
+      if: ${{ runner.os == 'Linux' && inputs.recent-cmake == 'false' }}
+      uses: jwlawson/actions-setup-cmake@802fa1a2c4e212495c05bf94dba2704a92a472be # v2.0.2
+      with:
+        cmake-version: '3.13.x'
+    - name: Set up CMake >= 3.18
+      if: ${{ runner.os == 'Linux' && inputs.recent-cmake == 'true' }}
+      uses: jwlawson/actions-setup-cmake@802fa1a2c4e212495c05bf94dba2704a92a472be # v2.0.2
+      with:
+        cmake-version: '3.18.x'
+    - name: Print CMake version
+      run: cmake --version
+      shell: bash
```

Comment:
Do we need to specify the shell? It should not matter what the shell is.

This coment also applies to the other "shell: bash" lines in this pull request except where noted.

Reply:
We actually do need to specify it in composite actions: https://github.com/orgs/community/discussions/18597